torch/_inductor/mkldnn_lowerings.py

```diff
@@ -544,24 +547,75 @@ def qlinear_unary(
+                    # but it is empty/dummy, which means the weight is symmetrically quantized.
+                    # We've already established by now that packed weights are int8, so if w_zp is not dummy,
+                    # then all its values must be zeros if weights are int8 symmetrically quantized.
+                    nonlocal w_zp
+                    if (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(w_zp),
+                            ir.ComputedBuffer,
+                        )
+                        and w_zp.get_numel() == 0
+                    ) or (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(w_zp),
+                            ir.ConstantBuffer,
+                        )
+                        and torch.equal(
+                            torch.zeros_like(V.graph.constants[w_zp.get_name()]),
+                            V.graph.constants[w_zp.get_name()],
+                        )
+                    ):
+                        return True
+                    else:
+                        return False
+
+                def _is_act_comptaible_with_onednn():
+                    # oneDNN doesn't currently have an optimized kernel for a vector activation zero-point.
+                    # So, if the activation zero point is a vector, we won't support it with oneDNN.
+                    # We can support the provided activation if:
+                    # 1. if x_zp is dummy (empty tensor) created to match an Inductor pattern
+                    # 2. If the activation is per-tensor quantized (x may even be uint8 in that case).
+                    # NOTE: since vector weight scale is supported with oneDNN qlinear in PyTorch,
+                    # we are able to support the case of per-token quantized activation & per-channel quantized weights
+                    # if x_zp is not present in a pattern, since in that case, we can apply the actual scale of x
+                    # after oneDNN qlinear in a replacement pattern. This is being done for a smoothquant pattern.
+                    # Inductor codegen is able to make it an epilogue fusion.
+                    # We don't support the case of per-token quantized activation with oneDNN qlinear if
+                    # the activation is quantized asymmetrically, or if the per-token quantized activation is
+                    # quantized symmetrically but still has a corresponding zero-points tensor,
+                    # which would be all zeros in case of int8).
+                    nonlocal x_zp
+                    if (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(x_zp),
+                            ir.ComputedBuffer,
+                        )
+                        and x_zp.get_numel() == 0
+                        and x.get_dtype() == torch.int8
```

Comment:
These 2 checks seems not correct. I suspect for the PT2E Dynamic Quant case, the x might with dtype `torch.uint8` and `x_zp.get_numel()` might be 1. cc @Xia-Weiwen

Reply:
There's no adverse side-effect since that scenario (dynamically quantized torch.uint8 activation) is not supported in the current main branch either. As discussed offline, we could support that case later. I'll add a comment. Thanks!

Reply:
Yes. `x_zp.get_numel()` can be zero.

Reply:
With more recent changes, PT2E dynamic per-tensor quantization case with torch.uint8 x would now be covered.